src/Entity/Programme.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace App\Entity;
+
+use Doctrine\Common\Collections\Collection;
+
+class Programme
+{
+    private int $id;
+
+    public string $name = '';
+
+    private \DateTime $startDate;
+
+    private \DateTime $endDate;
+
+    private ?User $trainer;
+
+    private Room $room;
+
+    private ?Collection $customers;
```

Comment:
why is this nullable?

Reply:
A program may have no customers in a specific moment (when created for example).
Also a program may have no trainer when it is imported... The trainer will be set up by management.
Programs without trainers should not be visible for apoinments.

Reply:
ok. but a programme with no customers means a program with an empty collection. An empty collection means a collection with no element, not a null object (think about the difference between an empty array and a null) :smiley: